src/azul/project/hca/transformers.py

```diff
@@ -13,12 +13,44 @@
 
 def _project_dict(bundle: api.Bundle) -> dict:
     project: api.Project
+    laboratories: Set[str]
+    institutions: Set[str]
+    contact_names: Set[str]
+    publication_titles: Set[str]
+
     project, *additional_projects = bundle.projects.values()
     reject(additional_projects, "Azul can currently only handle a single project per bundle")
+
+    # Store lists of all values of each of these facets to allow facet filtering
+    # and term counting on the webservice
+    laboratories = set()
+    institutions = set()
+    contact_names = set()
+    publication_titles = set()
+
+    for contributor in project.contributors:
```

Comment:
It really looks like you are extracting too many metadata properties in the metadata-api PR. The API should only expose the properties that are actually used.